mlflow/tracking/utils.py

```diff
@@ -70,8 +70,10 @@ def get_tracking_uri():
         return _tracking_uri
     elif env.get_env(_TRACKING_URI_ENV_VAR) is not None:
         return env.get_env(_TRACKING_URI_ENV_VAR)
-    else:
+    elif os.sep == "/":
```

Comment:
I tried that. Dealing with a tug and pull between the options right now.

Requiring the prefix lead to docker tests breaking because in many parts throughout the code base artifact_uri is used as a path, including project.run.

I think the end result will be removing those assumptions and using parse_path for each of those calls, or earlier in the stack like was done with self.artifact_dir.

However, changing all paths to absolute paths breaks a few of the other tests. I can make these updates but wanted to get a better idea on what the ideal end state would be.

Recent findings
- requiring file:/// broke the R implementation
- changing all paths to absolute_paths with uri of file:/// interacts poorly with tests, this can be mitigated, but most tests check for equality of passed in path and internal path(we can check abspath equality to mitigate)
- parsing logic will be simplified if we sanitize all entrypoints to file:///abs_path, currently it seems more complicated than before
- I have an edit to project.run related to this, where the path is parsed from the uri, similar changes will be needed but the basic idea is that the requirement touches most code paths but is basically to_uri or parse_path called on most uses of artfiact_uri.

I was adding backcompat support for linux paths to avoid enforcing that assumption but enforcing it earlier on may be better. 

Any thoughts @tomasatdatabricks

Reply:
Hi @eedeleon. Sorry for late response.

We can fix the tests, but let me first take a step back to make sure I understand the context here. What is the motivation to make the default uri on windows be file:///C/parent/child/? Would something break if the uri on windows was just a windows absolute path?

I think in general, we should have the file urls to work and I think it would be better to store the artifact uri in standardized format so I think your change makes sense.